zebra-network/src/protocol/external/codec.rs

```diff
@@ -103,6 +107,10 @@ impl Encoder for Codec {
         let mut body = Vec::new();
         self.write_body(&item, &mut body)?;
 
+        if body.len() >= self.builder.max_len {
```

Comment:
Issue: Messages that are exactly the maximum size are rejected, but they should be accepted.

Please fix the deserialisation case, and the serialisation case.

zcashd rejects messages greater than 2 MB:
https://github.com/zcash/zcash/blob/ca668cf666ec157bdecb019bf71803dbb19e7c07/src/net.cpp#L598
But accepts messages that are exactly 2 MB.

Reply:
Good one, fixed at https://github.com/ZcashFoundation/zebra/pull/645/commits/15245ad0256d6f177e873e5a737c6e8b80fdcc03